### PR TITLE
Init free changes

### PR DESCRIPTION
**Pull request by dnlmlr**
I also implemented  __attribute__((constructor)) and __attribute__((destructor)) for the InitEV3 and FreeEV3 functions but since you pushed the changes before me I rebased on your commit and just left the other changes I made in addition.

One commit is making the init and exit functions more consistent and the other commit renames the defined TRUE/FALSE to the normal true/false values from stdbool.

### CODE REVIEW

File: API/ev3_constants.h [hunk 1]
```diff
@@ -45,8 +45,6 @@ extern "C" {
 /** @addtogroup MiscConstants
  * @{
  */
-#define TRUE  true /*!< A true value */
```
Comment (a3f) [inline API/ev3_constants.h]: I would leave those definitions in for backwards compatibility.

File: API/ev3.c [hunk 2]
```diff
@@ -24,44 +24,22 @@ int __attribute__((constructor)) InitEV3 (void)
...
 	LcdInit();
-	LcdClean();
 	SoundInit();
-	initialized = true;
-
-	return 1;
-}
-
-
-int CloseEV3()
```
Comment (a3f) [inline API/ev3.c]: What if existing code already uses those? How about making `CloseEV3` and `FreeEV3` aliases for `ExitEV3` and make successive calls no-ops?

File: API/ev3_sensor.c [hunk 3]
```diff
@@ -101,11 +101,11 @@ int ir_sensor_channel[INPUTS];
 * 		 date: 2015-02-28
 *
 */
-int InitSensors()
+int SensorsInit()
 {
-	if (!EV3IsInitialized() && !InitEV3())
```
Comment (a3f) [inline API/ev3_sensor.c]: Good. Just noticed today, my commit would cause an infinite loop.. ^^'

File: API/ev3_sensor.h [hunk 4]
```diff
@@ -122,7 +134,7 @@ int SetIRBeaconCH(int sensorPort, int channel);
 /*!
  * @deprecated
 */
-#define initSensors() InitSensors()
```
Comment (a3f) [inline API/ev3_sensor.h]: Same. I am all for consistency, but maybe leave in an alias for backwards compatibility? @abke, any thoughts?

File: API/ev3_constants.h (hunk 1 above)
Comment (dnlmlr) [reply API/ev3_constants.h]: Yea this might indeed break some user code. I'll leave it in.

File: API/ev3.c (hunk 2 above)
Comment (dnlmlr) [reply API/ev3.c]: Calling only `CloseEV3` or `ExitEV3` would not properly shutdown the EV3 and calling both in the right order would be equivalent to `FreeEV3`. That's why I removed them.
Successive calls to the actual `Exit*` functions that are used in `ExitEV3` are no-ops already. 
Good point on compatibility though. I'll make the alias definitions.

File: API/ev3_sensor.h (hunk 4 above)
Comment (dnlmlr) [reply API/ev3_sensor.h]: Yea I think alias will be a good idea here.

File: API/ev3_constants.h (hunk 1 above)
Comment (dnlmlr) [reply API/ev3_constants.h]: Is in again

File: API/ev3.c (hunk 2 above)
Comment (dnlmlr) [reply API/ev3.c]: `CloseEV3` and `ExitEV3` now call `FreeEV3`

File: API/ev3_sensor.h (hunk 4 above)
Comment (dnlmlr) [reply API/ev3_sensor.h]: `InitSensors` and `initSensors` now both call the renamed `SensorsInit`

**Comment by a3f** [comment]
Thanks!

**Pull request merged**